src/solver/median/backward_median.cpp

```diff
@@ -0,0 +1,155 @@
+                                  const miopen::median::BwdProblemDescription& problem) const
+{
+    if(!IsImprovementOverROCm(problem))
+        return false;
+
+    if(!problem.IsValidFloat())
+        return false;
+
+    return true;
+}
+
+ConvSolution MedianBackward::GetSolution(const ExecutionContext& context,
+                                         const miopen::median::BwdProblemDescription& problem) const
+{
+    std::ignore = context;
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype    = problem.GetInputGradDesc().GetType();
+    auto io_dtype = miopen::GetDataType(dtype);
+
+    auto input_grad_lengths  = problem.GetInputGradDesc().GetLengths();
+    auto output_grad_lengths = problem.GetOutputGradDesc().GetLengths();
+
+    auto dim_size = input_grad_lengths[problem.GetDim()];
+
+    auto size        = problem.GetInputGradDesc().GetElementSize();
+    auto output_size = size / dim_size;
+
+    // Start building result.construction_params
+    size_t xlocalsize = LOCAL_SIZE;
+    size_t xgridsize  = output_size * xlocalsize;
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+    size_t zlocalsize = 1;
+    size_t zgridsize  = 1;
+
+    auto kernel        = KernelInfo{};
+    kernel.kernel_file = "MIOpenKthvalue.cpp";
+    kernel.kernel_name = "KthvalueBwd";
+
+    const auto build_params = KernelBuildParameters{
+        {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+        {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+        {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+        {"IN_OUT_TYPE", io_dtype == "bfloat16" ? "ushort" : io_dtype},
+        {"LOCAL_SIZE", xlocalsize},
+    };
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+    kernel.l_wk.push_back(zlocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+    kernel.g_wk.push_back(zgridsize);
+
+    result.construction_params.push_back(kernel);
+    // End building result.construction_params
+
+    // Start building result.invoker_factory
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params = raw_params.CastTo<miopen::median::BwdInvokeParams>();
+            size_t dim_stride     = params.inputGradDesc->GetStrides()[params.dim];
+
+            auto input_grad_tv = get_inner_expanded_tv<5>(deref(params.inputGradDesc));
+            auto input_grad_tv_without_selected_dim =
+                get_tv_without_dim<5>(input_grad_tv, params.dim);
+
+            auto output_grad_tv = get_inner_expanded_tv<5>(deref(params.outputGradDesc));
+            auto indices_tv     = get_inner_expanded_tv<5>(deref(params.indicesDesc));
```

Comment:
Let make number `5` be a constane of `#define`, for easier modification in the future. Check how did I do it with `VIEW_DIM` definition in [this](https://github.com/ROCm/MIOpen/blob/6cf29591839c05660ce4d0b6278123aa7b2dd0e7/src/solver/prelu/backward_prelu_multi_weights.cpp)

Reply:
solved